homeassistant/components/systemmonitor/repairs.py

```diff
@@ -51,7 +51,9 @@ async def async_step_migrate_process_sensor(
                 entity_reg.async_remove(entry.entity_id)
 
         self.hass.config_entries.async_update_entry(self.entry, options=new_options)
-        await self.hass.config_entries.async_reload(self.entry.entry_id)
+        self.hass.async_create_task(
+            self.hass.config_entries.async_reload(self.entry.entry_id)
+        )
```

Comment:
It would be use the new helper here in https://github.com/home-assistant/core/pull/110912 instead.

Reply:
Actually I think the test is wrong. Its now able to get reloaded in time so the entity gets created as unavailable instead of not being there.